io_scene_vox.py

```diff
@@ -203,52 +223,119 @@ def import_vox(path, *, voxel_spacing=1, voxel_size=1,
         for index in used_palette_indices:
             palette_entry = palette[index]
             material = bpy.data.materials.new("Voxel_mat{}".format(index))
-            material.diffuse_color = [pow(col / 255, gamma_value) for col in palette_entry[:3]]
-            material.diffuse_intensity = 1.0
-            material.alpha = palette_entry[3]
-            material.use_shadeless = use_shadeless
+            material.diffuse_color = [pow(col / 255, gamma_value) for col in palette_entry[:4]]
+            if use_shadeless:
+                material.use_nodes = True
+                material_diffuse_to_emission(material)
             mat_palette.update({index: material})
 
     # peel first voxel information
     voxel, *voxels = voxels
     location = [float(coord) * voxel_spacing for coord in voxel[:3]]
     # Using primitive_cube_add once here, to give us a template cube
-    bpy.ops.mesh.primitive_cube_add(radius=0.5 * voxel_size, location=location)
+    bpy.ops.mesh.primitive_cube_add(size=0.5 * voxel_size, location=location)
     base_voxel = bpy.context.object
     if use_palette:
         base_voxel.active_material = mat_palette[voxel[3]]
 
+    print("Loaded voxels:", len(voxels))
+
     to_link = []
 
     for voxel in voxels:
         copy = base_voxel.copy()
         copy.data = base_voxel.data.copy()
         copy.location = [float(coord) * voxel_spacing for coord in voxel[:3]]
+        #copy.parent = base_voxel
```

Comment:
```suggestion
```

Reply:
It was just a test to "hierarchical link" the base object you "clone" for each voxel. 
I think that base_voxel object should be deleted anyhow at the end of the import.

Reply:
I have had some thoughts about having a 'base model' be created, outside of the model, that all the voxels are cloned instances of. 

But that's beyond the scope of the port.

Currently it's commented out code, with no need to stick around